packages/core/src/node/messaging/ipc-bootstrap.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import 'reflect-metadata';
+import { ConsoleLogger } from 'vscode-ws-jsonrpc/lib/logger';
+import { createMessageConnection, IPCMessageReader, IPCMessageWriter, Trace } from 'vscode-jsonrpc';
+import { THEIA_PARENT_PID, THEIA_ENTRY_POINT, IPCEntryPoint } from './ipc-protocol';
+
+if (process.env[THEIA_PARENT_PID]) {
+    const parentPid = Number(process.env[THEIA_PARENT_PID]);
+
+    if (typeof parentPid === 'number' && !isNaN(parentPid)) {
+        setInterval(function () {
+            try {
+                // throws an exception if the main process doesn't exist anymore.
+                process.kill(parentPid, 0);
```

Comment:
`The ChildProcess object may emit an 'error' event if the signal cannot be delivered. Sending a signal to a child process that has already exited is not an error but may have unforeseen consequences. Specifically, if the process identifier (PID) has been reassigned to another process, the signal will be delivered to that process instead which can have unexpected results.`

From: https://nodejs.org/dist/latest-v8.x/docs/api/child_process.html#child_process_subprocess_kill_signal

Moreover 0 is not a valid signal number on linux at least, not sure what is supposed to happen.

The parent should just notify the child as it exists no ?

Reply:
I think similar work was done for the LSPs recently with onStop() ?

Reply:
this code is copied from vscode, they use it to ensure that child processes exit if the main process crashed and did not kill children, make sense only on windows, other OSs take care about child processes

Reply:
OK thx could you add a comment in the code about it please? it being windows specific?

Reply:
Note I don't think child processes exit when the parent exists on linux. You have to do it yourself like in the LSP case.

Reply:
will add, so far we experience it only on windows